lib/chef/provider/package/homebrew.rb

```diff
@@ -83,9 +87,40 @@ def brew(*args)
         # information, but that is not any more robust than using the
         # command-line interface that returns the same thing.
         #
-        # https://github.com/Homebrew/homebrew/wiki/Querying-Brew
+        # https://docs.brew.sh/Querying-Brew
+        #
+        # @returns [Hash] a hash of package information where the key is the package name
         def brew_info
-          @brew_info ||= Chef::JSONCompat.from_json(brew("info", "--json=v1", new_resource.package_name)).first
+          @brew_info ||= begin
+            command_array = ["info", "--json=v1"].concat Array(new_resource.package_name)
+            # convert the array of hashes into a hash where the key is the package name
+
+            cmd_output = brew_cmd_output(command_array, allow_failure: true)
+            return nil if cmd_output.empty? # empty std_out == bad package queried
+
+            Hash[Chef::JSONCompat.from_json(cmd_output).collect { |pkg| [pkg["name"], pkg] }]
+          end
+        end
+
+        #
+        # Return the package information given a package name or package alias
+        #
+        # @param [String] name_or_alias The name of the package or its alias
+        #
+        # @return [Hash] Package information
+        #
+        def package_info(package_name)
+          return nil if brew_info.nil? # continue to raise the nil up the chain
+
+          # return the package hash if it's in the brew info hash
+          return brew_info[package_name] if brew_info[package_name]
+
+          # check each item in the hash to see if we were passed an alias
+          brew_info.each_value do |p|
+            return p if p["aliases"].include?(package_name)
+          end
+
+          {}
```

Comment:
here you might actually want to return `{}` instead of nil for the package_info.

i didn't see this return value on first reading

Reply:
you could actually push this higher up and return an empty hash out of brew_info and just delete the nil check and it'll work fine here, that keeps the return signature of both packge_info and brew_info consistently returning a Hash and uses empty? instead of nil? as the guard condition.

Reply:
Done